### PR TITLE
DHFPROD-2222: Job endpoint returns null for id & Modify project watchers

### PR DESCRIPTION
**Pull request by hao1st**

### CODE REVIEW

File: web/src/main/java/com/marklogic/hub/web/web/CurrentProjectController.java [hunk 1]
```diff
@@ -300,8 +300,5 @@ private void clearAuthenticationAttributes(HttpServletRequest request) {
     @Override
     public void onLogoutSuccess(HttpServletRequest request, HttpServletResponse response, Authentication authentication) throws IOException, ServletException {
         request.getSession().invalidate();
-        String pluginDir = hubConfig.getHubPluginsDir().toString();
-        watcherService.removeListener(this);
-        watcherService.unwatch(pluginDir);
```
Comment (srinathgit) [inline web/src/main/java/com/marklogic/hub/web/web/CurrentProjectController.java]: Shouldn't we unwatch flows, stedef, entities dir snd also removelistener ?

Comment (hao1st) [reply web/src/main/java/com/marklogic/hub/web/web/CurrentProjectController.java]: @srinathgit add them.

File: web/src/main/java/com/marklogic/hub/web/web/CurrentProjectController.java [hunk 2]
```diff
@@ -230,15 +230,15 @@ public String preInstallCheck() {
...
         String flowsDir = hubConfig.getFlowsDir().toString();
         String stepDefinitionsDir = hubConfig.getStepDefinitionsDir().toString();
         String entitiesDir = hubConfig.getHubEntitiesDir().toString();
+        String mappingsDir = hubConfig.getHubMappingsDir().toString();
         if (!watcherService.hasListener(this)) {
-            watcherService.watch(pluginDir);
             watcherService.watch(flowsDir);
             watcherService.watch(stepDefinitionsDir);
             watcherService.watch(entitiesDir);
+            watcherService.watch(mappingsDir);
```
Comment (akshaysonvane) [inline web/src/main/java/com/marklogic/hub/web/web/CurrentProjectController.java]: Now that we are watching the entire mapping directory, should we remove the watch service from the mappingManagerService for the individual mapping folder name?

Comment (hao1st) [reply web/src/main/java/com/marklogic/hub/web/web/CurrentProjectController.java]: We can leave them there now. Let us test after beta release.

**Comment by akshaysonvane** [comment]
@aebadirad I was under the impression that we need watcher service on the plugins dir as the 4.x flows still reside there. Has that been changed?

**Comment by aebadirad** [comment]
We just deploy & run them, but we offer no way to interact with the 4.x /plugins currently. So the watcher isn't needed once an upgrade has been completed. I'd rather not keep an active watch that's 'hidden' from the user, giving them unexpected behavior.

File: web/src/main/java/com/marklogic/hub/web/web/CurrentProjectController.java (hunk 1 above)
Comment (aebadirad) [reply web/src/main/java/com/marklogic/hub/web/web/CurrentProjectController.java]: Yes we should also unwatch them to clear it.

**Pull request merged**